### PR TITLE
Merge RNNPool Codes

### PR DESCRIPTION
**Pull request by oindrilasaha**
Add face detection, visual wakeword examples

### CODE REVIEW

File: examples/pytorch/FastCells/fastcell_example.py [hunk 1]
```diff
@@ -76,6 +76,8 @@ def main():
     else:
         sys.exit('Exiting: No Such Cell as ' + cell)
 
+    # import pdb;pdb.set_trace()
+
```
Comment (adityakusupati) [inline examples/pytorch/FastCells/fastcell_example.py:80]: Remove the comment.

File: examples/pytorch/vision/Face_Detection/data/widerface.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        #img = Image.fromarray(img)
+        '''
+        draw = ImageDraw.Draw(img)
+        w,h = img.size
+        for bbox in sample_labels:
+            bbox = (bbox[1:] * np.array([w, h, w, h])).tolist()
+
+            draw.rectangle(bbox,outline='red')
+        img.save('image.jpg')
+        '''
```
Comment (adityakusupati) [inline examples/pytorch/vision/Face_Detection/data/widerface.py:88]: Remove the comments.

File: pytorch/edgeml_pytorch/graph/rnnpool.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        torch.zeros(2, batch_size, self.nHiddenDimsBiDir).to(torch.device("cuda")),
+                        torch.zeros(2, batch_size, self.nHiddenDimsBiDir).to(torch.device("cuda")))
+
+        # stateList_reverse = tuple(reversed(stateList))
+
+        # outputs_reverse = self.cell_bidirrnn(torch.stack(stateList_reverse),
+        #                 (torch.randn(batch_size, self.nHiddenDimsBiDir).to(torch.device("cuda")),
+        #                 torch.randn(batch_size, self.nHiddenDimsBiDir).to(torch.device("cuda"))))  
+
+        # finalHidden2 = torch.cat([outputs[-1],outputs_reverse[-1]],1)
```
Comment (adityakusupati) [inline pytorch/edgeml_pytorch/graph/rnnpool.py:73]: Remove comments.

File: pytorch/edgeml_pytorch/trainer/fastTrainer.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ def __init__(self, FastObj, numClasses, sW=1.0, sU=1.0,
         self.sU = sU
 
         self.numClasses = numClasses
-        self.inputDims = self.FastObj.input_size
+        # self.inputDims = self.FastObj.input_size
```
Comment (adityakusupati) [inline pytorch/edgeml_pytorch/trainer/fastTrainer.py:31]: remove the comment.

File: pytorch/edgeml_pytorch/trainer/fastTrainer.py [hunk 5]
```diff
@@ -391,6 +393,8 @@ def train(self, batchSize, totalEpochs, Xtrain, Xtest, Ytrain, Ytest,
                 k = shuffled[j * batchSize:(j + 1) * batchSize]
                 batchX = Xtrain[:, k, :]
                 batchY = Ytrain[k]
+                # import pdb;pdb.set_trace()
```
Comment (adityakusupati) [inline pytorch/edgeml_pytorch/trainer/fastTrainer.py:396]: remove the comment.

File: pytorch/edgeml_pytorch/trainer/fastTrainer.py [hunk 6]
```diff
@@ -444,7 +448,7 @@ def train(self, batchSize, totalEpochs, Xtrain, Xtest, Ytrain, Ytest,
                 if maxTestAcc <= testAcc:
                     maxTestAccEpoch = i
                     maxTestAcc = testAcc
-                    self.saveParams(currDir)
+                    # self.saveParams(currDir)
```
Comment (adityakusupati) [inline pytorch/edgeml_pytorch/trainer/fastTrainer.py:451]: Why did we comment this out?

**Review by harsha-simhadri — COMMENTED** [review]
Could you please:
1. move this file to git lfs, its 6+MB: examples/pytorch/vision/cpp/trace_0_input.npy
2. Could you please remove this empty file: examples/pytorch/vision/cpp/trace_0_output.npy

**Review by harsha-simhadri — APPROVED** [review]

**Pull request merged**